v bug   compiler bug

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** 
**OS:** 

**OS: windows, Microsoft Windows 10 Pro v19044 64-bit
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i7-4600U CPU @ 2.10GHz  
CC version: Error: exec failed (CreateProcess) with code 2: The system cannot find the file specified.
 cmd: cc --version

getwd: C:\Users\mlap\Documents\prbs\vpros
vmodules: C:\Users\mlap\.vmodules
vroot: C:\msys64\home\mlap\v
vexe: C:\msys64\home\mlap\v\v.exe
vexe mtime: 2022-08-06 05:06:05
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.0 372857c.095f4bc

Git version: git version 2.27.0.windows.1
Git vroot status: weekly.2022.31-19-g095f4bcf (21 commit(s) behind V master)
.git/config present: true
thirdparty/tcc status: thirdparty-windows-amd64 125e21e2**

<!-- Please include the standalone minimal reproducible code.
If not, it may be closed by the administrators. -->
**What did you do?**

```v
module main

fn main() {
    
    mut xa := fn (x fn (int) string, y int) string {return x(y)}
    a := xa(fn (i int) string {return 'a'+i.str()}, 8)
    println(a)
}
```

**What did you expect to see?**
print string
a8
 
**What did you see instead?**
```
==================
C:/Users/mlap/AppData/Local/Temp/v_0/a.13552825952583261223.tmp.c:462: warning: WINVER redefined
C:/Users/mlap/AppData/Local/Temp/v_0/a.13552825952583261223.tmp.c:1917: error: ';' expected (got "string_int__string_145")       
...
==================
(Use `v -cg` to print the entire error message)

builder error:
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

--- a/vlib/v/ast/table.v
+++ b/vlib/v/ast/table.v
@@ -219,7 +219,7 @@ pub fn (t &Table) fn_type_signature(f &Fn) string {
 		} else {
 			sig += arg_type_sym.str().to_lower().replace_each(['.', '__', '&', '', '[', 'arr_',
 				'chan ', 'chan_', 'map[', 'map_of_', ']', '_to_', '<', '_T_', ',', '_', ' ', '',
-				'>', ''])
+				'>', '', '(', '_', ')', '_'])
 		}
 		if i < f.params.len - 1 {
 			sig += '_'

--- a/vlib/v/tests/anon_fn_with_nested_anon_fn_args_test.v
+++ b/vlib/v/tests/anon_fn_with_nested_anon_fn_args_test.v
@@ -1,0 +1,12 @@
+module main
+
+fn test_anon_fn_with_nested_anon_fn_args() {
+	mut xa := fn (x fn (int) string, y int) string {
+		return x(y)
+	}
+	a := xa(fn (i int) string {
+		return 'a' + i.str()
+	}, 8)
+	println(a)
+	assert a == 'a8'
+}

